bundles/org.openhab.binding.monopriceaudio/src/main/java/org/openhab/binding/monopriceaudio/internal/communication/MonopriceAudioReaderThread.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.monopriceaudio.internal.communication;
+
+import java.util.Arrays;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.monopriceaudio.internal.MonopriceAudioBindingConstants;
+import org.openhab.binding.monopriceaudio.internal.MonopriceAudioException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A class that reads messages from the MonopriceAudio device in a dedicated thread
+ *
+ * @author Laurent Garnier - Initial contribution
+ * @author Michael Lobstein - Adapted for the MonopriceAudio binding
+ */
+@NonNullByDefault
+public class MonopriceAudioReaderThread extends Thread {
+    private static final int READ_BUFFER_SIZE = 16;
+    private static final int SIZE = 64;
+    private static final char TERM_CHAR = '\r';
+    private final Logger logger = LoggerFactory.getLogger(MonopriceAudioReaderThread.class);
+
+    private MonopriceAudioConnector connector;
+
+    /**
+     * Constructor
+     *
+     * @param connector the object that should handle the received message
+     */
+    public MonopriceAudioReaderThread(MonopriceAudioConnector connector) {
+        super(MonopriceAudioBindingConstants.BINDING_ID);
+        this.connector = connector;
+    }
```

Comment:
Make sure that any created threads are daemon.
```suggestion
        super(MonopriceAudioBindingConstants.BINDING_ID);
        this.connector = connector;
        setDaemon(true);
    }
```